nostr/save_nostr_key.js

```diff
@@ -0,0 +1,132 @@
+const asyncAuto = require('async/auto');
+const {returnResult} = require('asyncjs-util');
+
+const {encryptToNode} = require('../encryption');
+const {homePath} = require('../storage');
+
+const defaultRelaysFile = {nostr: []};
+const nostrKeyFilePath = () => homePath({file: 'nostr.json'}).path;
+const {parse} = JSON;
+const stringify = obj => JSON.stringify(obj, null, 2);
+
+
+/** Save encrypted nostr private key
+
+  {
+    fs: {
+      getFile: <Read File Contents Function> (path, cbk) => {}
+      makeDirectory: <Make Directory Function> (path, cbk) => {}
+      writeFile: <Write File Contents Function> (path, contents, cbk) => {}
+    }
+    key: <Nostr Private Key String>
+    lnd: <Authenticated LND API Object>
+    logger: <Winston Logger Object>
+    node: <Saved Node Name String>
+  }
+
+  @returns via cbk or Promise
+*/
+module.exports = (args, cbk) => {
+  return new Promise((resolve, reject) => {
+    return asyncAuto({
+      // Check arguments
+      validate: cbk => {
+        if (!args.fs) {
+          return cbk([400, 'ExpectedFilesystemMethodsToSaveNostrKey']);
+        }
+
+        if (!args.key) {
+          return cbk([400, 'ExpectedNostrPrivateKeyToSaveNostrKey']);
+        }
+
+        if (!args.lnd) {
+          return cbk([400, 'ExpectedLndToSaveNostrKey']);
+        }
+
+        if (!args.logger) {
+          return cbk([400, 'ExpectedLoggerToSaveNostrKey']);
+        }
+
+        return cbk();
+      },
+
+      // Register the home directory
+      registerHomeDir: ['validate', ({}, cbk) => {
+        return args.fs.makeDirectory(homePath({}).path, err => {
+          // Ignore errors, the directory may already be there
+          return cbk();
+        });
+      }],
+
+      // Encrypt the private key
+      encryptKey: ['validate', ({}, cbk) => {
+        return encryptToNode({
+          lnd: args.lnd,
+          message: args.key,
+        },
+        cbk)
+      }],
+
+      readFile: ['encryptKey', 'validate', ({encryptKey}, cbk) => {
+        const node = args.node || "";
+        
+        return args.fs.getFile(nostrKeyFilePath(), (err, res) => {
+          // Ignore errors, the file may not exist
+          if (!!err || !res) {
+            defaultRelaysFile.nostr.push({key: encryptKey.encrypted, node, relays: []});
```

Comment:
i think this should be a copy of the default relays file, or define it in full here